rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -5097,7 +5070,8 @@ static void rmw_fini_cs(CddsCS * cs)
   dds_delete(cs->pub->enth);
 }
 
-static rmw_ret_t destroy_client(const rmw_node_t * node, rmw_client_t * client)
+static rmw_ret_t destroy_client(
+  const rmw_node_t * node, rmw_client_t * client, bool destroy_graph = true)
```

Comment:
Can you please describe why we need the `destroy_graph` boolean here?  It isn't totally clear to me.

Reply:
After using `make_scope_exit` in the `rmw_create_client`, I removed the `destroy_client` and move its original implementation code into `rmw_destroy_client since `destroy_client` is only called once. There is no boolean flag.